mlflow/tensorflow.py

```diff
@@ -989,110 +985,118 @@ def _log_early_stop_callback_metrics(callback, history, metrics_logger):
                     last_epoch = len(history.history[metric_key])
                     metrics_logger.record_metrics(restored_metrics, last_epoch)
 
-    def fit(self, *args, **kwargs):
-        with _manage_active_run() as run:
-            original = gorilla.get_original_attribute(tensorflow.keras.Model, "fit")
-
-            unlogged_params = ["self", "x", "y", "callbacks", "validation_data", "verbose"]
-
-            log_fn_args_as_params(original, args, kwargs, unlogged_params)
-            early_stop_callback = None
-
-            run_id = run.info.run_id
-            with batch_metrics_logger(run_id) as metrics_logger:
-                # Checking if the 'callback' argument of fit() is set
-                if len(args) >= 6:
-                    tmp_list = list(args)
-                    early_stop_callback = _early_stop_check(tmp_list[5])
-                    tmp_list[5], log_dir = _setup_callbacks(tmp_list[5], log_models, metrics_logger)
-                    args = tuple(tmp_list)
-                elif kwargs.get("callbacks"):
-                    early_stop_callback = _early_stop_check(kwargs["callbacks"])
-                    kwargs["callbacks"], log_dir = _setup_callbacks(
-                        kwargs["callbacks"], log_models, metrics_logger
-                    )
-                else:
-                    kwargs["callbacks"], log_dir = _setup_callbacks([], log_models, metrics_logger)
+    def fit(original, self, *args, **kwargs):
+        active_run = mlflow.active_run()
+        if MLFLOW_AUTOLOGGING in active_run.data.tags:
+            global _AUTOLOG_RUN_ID
+            _AUTOLOG_RUN_ID = active_run.info.run_id
```

Comment:
If the current active run was generated by autologging, store the run_id in `global _AUTOLOG_RUN_ID`.

Reply:
What happens if a user creates their own run and performs autologging within it? Looks like we won't capture model artifacts in that case. I think this should just be `_AUTOLOG_RUN_ID = mlflow.active_run().info.run_id`

Reply:
If a user runs:

```
with mlflow.start_run():
    model.train(...)
    model.export_saved_model()
```

`export_saved_model` works like:

```python
    def export_saved_model(original, self, *args, **kwargs):
        auto_end = False
        if not mlflow.active_run():  # not mlflow.active_run() returns False
            global _AUTOLOG_RUN_ID
            if _AUTOLOG_RUN_ID:
                try_mlflow_log(mlflow.start_run, _AUTOLOG_RUN_ID)
            else:
                try_mlflow_log(mlflow.start_run)
                try_mlflow_log(mlflow.set_tag, MLFLOW_AUTOLOGGING, FLAVOR_NAME)
                auto_end = True

        serialized = original(self, *args, **kwargs)
        try_mlflow_log(...)  # logs the model in the active run

        if (
            mlflow.active_run() is not None and
            mlflow.active_run().info.run_id == _AUTOLOG_RUN_ID  # this return False
        ) or auto_end:
            try_mlflow_log(mlflow.end_run)
        return serialized
```

Reply:
Ah, got it. Totally missed the `if not mlflow.active_run()`. All good :)! Thanks for clarifying!